Someone has reported a crash for `ZimFileReader.getFavicon`.

**Describe the bug**
Someone has reported a crash for `ZimFileReader.getFavicon`.

**Expected behavior**
It should not crash the application; instead, if there is some error, it should catch that error and work normally.

**Environment**
- Version of Kiwix Android: 3.14.0 [7231312]
- Device : SV55216
- OS version : Android 11

**Logs**
```kotlin
Hi Kiwix Developers!
The Android app crashed, here are some details to help fix it:
   
      Exception Details:
    java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
Caused by: java.lang.reflect.InvocationTargetException
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
... 1 more
Caused by: java.lang.Exception:
Assertion failed at ../../SOURCE/libzim_release/src/file_reader.cpp:298
offset.v+size.v[4294967296] <= _size.v[2773471377]
at org.kiwix.libzim.Item.getData(Native Method)
at org.kiwix.kiwixmobile.core.reader.ZimFileReader.getFavicon(SourceFile:15)
at org.kiwix.kiwixmobile.core.main.CoreWebViewClient.onPageFinished(SourceFile:341)
at WV.r23.e(chromium-TrichromeWebViewGoogle.aab-stable-739012430:12)
at WV.ch.handleMessage(chromium-TrichromeWebViewGoogle.aab-stable-739012430:250)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:223)
at android.app.ActivityThread.main(ActivityThread.java:7725)
... 3 more

  Current Zim File:
null
All Zim Files in DB:


  Current Locale:
ar

  BluetoothClass.Device Details:
Device:[SV55216]
Model:[Schok Volt SV55]
Manufacturer:[Schok]
Time:[1698897827000]
Android Version:[11]
App Version:[3.14.0 7231312]

      Mount Points
    MountInfo(device=/dev/block/dm-4, mountPoint=/, fileSystem=ext4)
, MountInfo(device=tmpfs, mountPoint=/dev, fileSystem=tmpfs)
, MountInfo(device=devpts, mountPoint=/dev/pts, fileSystem=devpts)
, MountInfo(device=none, mountPoint=/dev/blkio, fileSystem=cgroup)
, MountInfo(device=none, mountPoint=/dev/cpuctl, fileSystem=cgroup)
, MountInfo(device=none, mountPoint=/dev/cpuset, fileSystem=cgroup)
, MountInfo(device=none, mountPoint=/dev/memcg, fileSystem=cgroup)
, MountInfo(device=none, mountPoint=/dev/stune, fileSystem=cgroup)
, MountInfo(device=binder, mountPoint=/dev/binderfs, fileSystem=binder)
, MountInfo(device=adb, mountPoint=/dev/usb-ffs/adb, fileSystem=functionfs)
, MountInfo(device=diag, mountPoint=/dev/ffs-diag, fileSystem=functionfs)
, MountInfo(device=diag_mdm, mountPoint=/dev/ffs-diag-1, fileSystem=functionfs)
, MountInfo(device=diag_mdm2, mountPoint=/dev/ffs-diag-2, fileSystem=functionfs)
, MountInfo(device=proc, mountPoint=/proc, fileSystem=proc)
, MountInfo(device=sysfs, mountPoint=/sys, fileSystem=sysfs)
, MountInfo(device=selinuxfs, mountPoint=/sys/fs/selinux, fileSystem=selinuxfs)
, MountInfo(device=none, mountPoint=/sys/fs/cgroup, fileSystem=cgroup2)
, MountInfo(device=tracefs, mountPoint=/sys/kernel/tracing, fileSystem=tracefs)
, MountInfo(device=none, mountPoint=/sys/fs/fuse/connections, fileSystem=fusectl)
, MountInfo(device=bpf, mountPoint=/sys/fs/bpf, fileSystem=bpf)
, MountInfo(device=tmpfs, mountPoint=/mnt, fileSystem=tmpfs)
, MountInfo(device=tmpfs, mountPoint=/mnt/installer, fileSystem=tmpfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/installer/0/emulated, fileSystem=fuse)
, MountInfo(device=/data/media, mountPoint=/mnt/installer/0/emulated/0/Android/data, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/installer/0/emulated/0/Android/obb, fileSystem=sdcardfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/installer/0/CC26-36AF, fileSystem=fuse)
, MountInfo(device=tmpfs, mountPoint=/mnt/androidwritable, fileSystem=tmpfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/androidwritable/0/emulated, fileSystem=fuse)
, MountInfo(device=/data/media, mountPoint=/mnt/androidwritable/0/emulated/0/Android/data, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/androidwritable/0/emulated/0/Android/obb, fileSystem=sdcardfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/androidwritable/0/CC26-36AF, fileSystem=fuse)
, MountInfo(device=/dev/block/bootdevice/by-name/persist, mountPoint=/mnt/vendor/persist, fileSystem=ext4)
, MountInfo(device=/dev/block/bootdevice/by-name/persist, mountPoint=/mnt/umx, fileSystem=ext4)
, MountInfo(device=/data/media, mountPoint=/mnt/runtime/default/emulated, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/runtime/read/emulated, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/runtime/write/emulated, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/runtime/full/emulated, fileSystem=sdcardfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/user/0/emulated, fileSystem=fuse)
, MountInfo(device=/data/media, mountPoint=/mnt/user/0/emulated/0/Android/data, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/user/0/emulated/0/Android/obb, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/mnt/pass_through/0/emulated, fileSystem=sdcardfs)
, MountInfo(device=/dev/block/vold/public:179,33, mountPoint=/mnt/media_rw/CC26-36AF, fileSystem=vfat)
, MountInfo(device=/mnt/media_rw/CC26-36AF, mountPoint=/mnt/runtime/default/CC26-36AF, fileSystem=sdcardfs)
, MountInfo(device=/mnt/media_rw/CC26-36AF, mountPoint=/mnt/runtime/read/CC26-36AF, fileSystem=sdcardfs)
, MountInfo(device=/mnt/media_rw/CC26-36AF, mountPoint=/mnt/runtime/write/CC26-36AF, fileSystem=sdcardfs)
, MountInfo(device=/mnt/media_rw/CC26-36AF, mountPoint=/mnt/runtime/full/CC26-36AF, fileSystem=sdcardfs)
, MountInfo(device=/dev/fuse, mountPoint=/mnt/user/0/CC26-36AF, fileSystem=fuse)
, MountInfo(device=/mnt/media_rw/CC26-36AF, mountPoint=/mnt/pass_through/0/CC26-36AF, fileSystem=sdcardfs)
, MountInfo(device=/dev/block/by-name/metadata, mountPoint=/metadata, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-5, mountPoint=/product, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-6, mountPoint=/vendor, fileSystem=ext4)
, MountInfo(device=/dev/block/bootdevice/by-name/modem, mountPoint=/vendor/firmware_mnt, fileSystem=vfat)
, MountInfo(device=/dev/block/bootdevice/by-name/dsp, mountPoint=/vendor/dsp, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-7, mountPoint=/system_ext, fileSystem=ext4)
, MountInfo(device=tmpfs, mountPoint=/apex, fileSystem=tmpfs)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.mediaprovider, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.sdkext, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.tethering, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.wifi, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.resolv, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.adbd, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.permission, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.runtime, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.media.swcodec, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.neuralnetworks, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.media, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.extservices, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.conscrypt, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.os.statsd, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.i18n, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.vndk.v30, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.ipsec, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.art, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-4, mountPoint=/apex/com.android.tzdata, fileSystem=ext4)
, MountInfo(device=tmpfs, mountPoint=/linkerconfig, fileSystem=tmpfs)
, MountInfo(device=tmpfs, mountPoint=/linkerconfig, fileSystem=tmpfs)
, MountInfo(device=none, mountPoint=/acct, fileSystem=cgroup)
, MountInfo(device=none, mountPoint=/config, fileSystem=configfs)
, MountInfo(device=/dev/block/bootdevice/by-name/cache, mountPoint=/cache, fileSystem=ext4)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/user/0, fileSystem=f2fs)
, MountInfo(device=tmpfs, mountPoint=/data_mirror, fileSystem=tmpfs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data_mirror/data_ce/null, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data_mirror/data_ce/null/0, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data_mirror/data_de/null, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data_mirror/cur_profiles, fileSystem=f2fs)
, MountInfo(device=tmpfs, mountPoint=/storage, fileSystem=tmpfs)
, MountInfo(device=/dev/fuse, mountPoint=/storage/emulated, fileSystem=fuse)
, MountInfo(device=/data/media, mountPoint=/storage/emulated/0/Android/data, fileSystem=sdcardfs)
, MountInfo(device=/data/media, mountPoint=/storage/emulated/0/Android/obb, fileSystem=sdcardfs)
, MountInfo(device=/dev/fuse, mountPoint=/storage/CC26-36AF, fileSystem=fuse)
, MountInfo(device=tmpfs, mountPoint=/data/data, fileSystem=tmpfs)
, MountInfo(device=tmpfs, mountPoint=/data/user, fileSystem=tmpfs)
, MountInfo(device=tmpfs, mountPoint=/data/user_de, fileSystem=tmpfs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/user_de/0/org.kiwix.kiwixmobile, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/data/org.kiwix.kiwixmobile, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/user_de/0/com.google.android.gms, fileSystem=f2fs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/data/com.google.android.gms, fileSystem=f2fs)
, MountInfo(device=tmpfs, mountPoint=/data/misc/profiles/cur, fileSystem=tmpfs)
, MountInfo(device=/dev/block/dm-8, mountPoint=/data/misc/profiles/cur/0/org.kiwix.kiwixmobile, fileSystem=f2fs)

    External Directories
    /storage/emulated/0/Android/data/org.kiwix.kiwixmobile/files
/storage/CC26-36AF/Android/data/org.kiwix.kiwixmobile/files
```

--- a/core/src/main/java/org/kiwix/kiwixmobile/core/reader/ZimFileReader.kt
+++ b/core/src/main/java/org/kiwix/kiwixmobile/core/reader/ZimFileReader.kt
@@ -57,7 +57,6 @@ import javax.inject.Inject
 
 private const val TAG = "ZimFileReader"
 
-@Suppress("LongParameterList")
 class ZimFileReader constructor(
   val zimReaderSource: ZimReaderSource,
   val jniKiwixReader: Archive,
@@ -150,15 +149,14 @@ class ZimFileReader constructor(
   val date: String get() = getSafeMetaData("Date", "")
   val description: String get() = getSafeMetaData("Description", "")
   val favicon: String?
-    get() =
-      if (jniKiwixReader.hasIllustration(ILLUSTRATION_SIZE)) {
-        Base64.encodeToString(
-          jniKiwixReader.getIllustrationItem(ILLUSTRATION_SIZE).data.data,
-          Base64.DEFAULT
-        )
-      } else {
-        null
-      }
+    get() = runCatching {
+      Base64.encodeToString(
+        jniKiwixReader.getIllustrationItem(ILLUSTRATION_SIZE).data.data,
+        Base64.DEFAULT
+      )
+    }.onFailure {
+      Log.e(TAG, "Could not get the favicon for $title. Original exception: $it")
+    }.getOrNull()
   val language: String get() = getSafeMetaData("Language", "")
 
   val tags: String

